### ISSUE TITLE
Timeout Issues

### ISSUE THREAD

**Issue by DanielRiveraHQ** [algolia/gatsby-plugin-algolia#117]
Hey guys, just wondering if there is any way I can set something to increase the time to declare timeout. 

I have a big website with lots of index and my production build is failing 

Thank you for your help

**Comment by DanielRiveraHQ** [comment]
Adding a screenshot of the log I'm getting:

![Screen Shot 2021-02-21 at 3 44 05 PM](https://user-images.githubusercontent.com/20131939/108640282-6084dd00-745e-11eb-8e8a-30cfe1b91898.png)

Do you think it can be customized in the plugin the timeout accessing this?:
https://www.algolia.com/doc/api-reference/api-methods/configuring-timeouts/

**Comment by eunjae-lee** [comment]
@Haroenv I think the error is thrown from around [here](https://github.com/algolia/algoliasearch-client-javascript/blob/3.24.5/src/browser/createAlgoliasearch.js#L166-L171).
Do you see any solution or workaround? Only a way in my mind is to patch-package the timeout values within algoliasearch.

**Comment by Haroenv** [comment]
Could you change 

https://github.com/algolia/gatsby-plugin-algolia/blob/2f4c093e90e128e351fd77b2a8195811cf0f6026/gatsby-node.js#L46

to:

```js
 const client = algoliasearch(appId, apiKey, { timeout: 30_000 }); 
```

and see if this makes a difference? Does this consistently happen for you @DanielRiveraHQ

**Comment by DanielRiveraHQ** [comment]
@Haroenv @eunjae-lee This worked great! I had to implement another fix about the new method to use GraphQL in Gatsby. 

Here is the forked repo I created in case you want to take a look: https://github.com/Revere-Health/gatsby-plugin-algolia

**Comment by Haroenv** [comment]
this has been fixed in https://github.com/algolia/gatsby-plugin-algolia/commit/7641b45b8123aaca423b2c588b7d200dfeb0c3da and published in 0.16.5!
